File: src/main/java/com/dyonovan/jatm/collections/DummyState.java
package com.dyonovan.jatm.collections;

import com.dyonovan.jatm.common.blocks.BlockBakeable;
import com.google.common.collect.ImmutableMap;
import net.minecraft.block.Block;
import net.minecraft.block.properties.IProperty;
import net.minecraft.block.state.IBlockState;
import net.minecraft.util.BlockPos;
import net.minecraft.world.IBlockAccess;

import java.util.Collection;

public class DummyState implements IBlockState {
    public IBlockAccess blockAccess;
    public BlockPos pos;
    public BlockBakeable block;

    public DummyState(IBlockAccess w, BlockPos p, BlockBakeable blockBakeable) {
        this.blockAccess = w;
        this.pos = p;
        this.block = blockBakeable;
    }

    public Collection getPropertyNames() {
        return null;
    }

    public Comparable getValue(IProperty property) {
        return null;
    }

    public IBlockState withProperty(IProperty property, Comparable value) {
        return null;
    }

    public IBlockState cycleProperty(IProperty property) {
        return null;
    }

    public ImmutableMap getProperties() {
        return null;
    }

    public BlockBakeable getBlock() {
        return block;
    }
}

File: src/main/scala/com/dyonovan/neotech/collections/DummyState.scala
package com.dyonovan.neotech.collections

import java.util

import com.google.common.collect.ImmutableMap
import com.teambr.bookshelf.common.blocks.traits.BlockBakeable
import net.minecraft.block.Block
import net.minecraft.block.properties.IProperty
import net.minecraft.block.state.IBlockState
import net.minecraft.util.BlockPos
import net.minecraft.world.IBlockAccess

/**
 * This file was created for NeoTech
 *
 * NeoTech is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Dyonovan
 * @since August 17, 2015
 */
class DummyState(w: IBlockAccess, p: BlockPos, blockBakeable: BlockBakeable) extends IBlockState {

    val blockAccess = w
    val pos = p
    val block = blockBakeable

    override def withProperty(property: IProperty, value: Comparable[_]): IBlockState = null

    override def getValue(property: IProperty): Comparable[_] = null

    override def cycleProperty(property: IProperty): IBlockState = null

    override def getPropertyNames: util.Collection[_] = null

    override def getBlock: Block = block

    override def getProperties: ImmutableMap[_, _] = null
}
